api/app/controllers/webhook.controller.ts

```diff
@@ -174,109 +178,120 @@ async function processCommit(
 ) {
   const changedFiles = [...commit[type]]; // get the files that were added or modified in the commit so we don't have to maintain two separate functions, this is also easily extendable when we want to take care of deleted commits.
   for (const file of changedFiles) {
-    const rawUrl = `https://api.github.com/repos/${pushEvent.repository?.full_name}/contents/${file}?ref=${branch}`;
-    const response = await axios.get(rawUrl, {
-      headers: { Accept: "application/vnd.github.v3.raw" },
-    });
-    const mdContent = response.data;
-    const jsonContent: BaseParsedMdContent =
-      parseMdToJSON<BaseParsedMdContent>(mdContent);
-    const transcript_by = jsonContent.transcript_by.toLowerCase();
-
-    function isTranscriptValid(transcript_by: string): boolean {
-      const lowerCaseTranscriptBy = transcript_by.toLowerCase();
-      return (
-        lowerCaseTranscriptBy.includes("tstbtc") &&
-        lowerCaseTranscriptBy.includes("--needs-review")
-      );
-    }
-
-    if (!jsonContent) {
-      throw new Error(
-        "Malformed data: transcript content might not be in the correct format"
-      );
-    }
+    try {
+      const rawUrl = `https://api.github.com/repos/${pushEvent.repository?.full_name}/contents/${file}?ref=${branch}`;
+      const response = await axios.get(rawUrl, {
+        headers: { Accept: "application/vnd.github.v3.raw" },
+      });
+      const mdContent = response.data;
+      const jsonContent: BaseParsedMdContent =
+        parseMdToJSON<BaseParsedMdContent>(mdContent);
+      const transcript_by = jsonContent.transcript_by.toLowerCase();
+
+      function isTranscriptValid(transcript_by: string): boolean {
```

Comment:
this function should be in utils for transcripts not defined inline the loop.

Reply:
fixed

Reply:
this hasn't been fixed. can't find the commit that fixes it